mango-python/bdgenomics/mango/variants.py

```diff
@@ -76,3 +83,165 @@ def viewPileup(self, contig, start, end, reference = 'hg19', label = "Variants",
             locus="%s:%i-%i" % (contig, start, end)
             return pileup.PileupViewer(locus=locus, reference=reference, tracks=tracks)
 
+
+class VariantsPerSampleDistribution(object):
+    """ VariantsPerSampleDistribution class.
+    VariantsPerSampleDistribution computes a distribution of the count of variants per sample.
+    """
+    def __init__(self,ss,genotypeRDD):
+        """
+        Initializes a VariantsPerSampleDistribution class.
+        Computes the counts of variants per sample from a genotypeRDD.
+
+        :param ss: SparkContext
+        :param genotypeRDD: bdgenomics genotypeRDD
+        """
+        df =  genotypeRDD.toDF()
+        data =  df.where( (df.alleles[0] == u'ALT') | (df.alleles[1] == u'ALT')).groupBy("sampleid").count().collect()
```

Comment:
Does `.count `call a collection?

Reply:
`.count` here returns the count of items per group for a collection that has already been grouped.  I believe it is this:
```
https://spark.apache.org/docs/latest/api/python/pyspark.sql.html#pyspark.sql.functions.count
```
A similar dataframe query in the callrate function 
```
called_df =  df.where( (df.alleles[0] != u'NO_CALL')).groupBy("sampleid").count().collect()
```
returns final  from collected return value:
```
[Row(sampleid=u'7', count=604),
 Row(sampleid=u'51', count=775),
 Row(sampleid=u'15', count=897),
```

Reply:
This seems pretty inefficient. There is a 3 calls that shuffle/collect data in a row.

Reply:
This was running for 2.6 hours on a single chromosome (chr1 from 1000g).  I don't think this approach is going to be performant enough.

Reply:
Its unfortunate if it is shuffling as the dataframe GroupBy aggregation function claims to be using a hash when it can assume that they key space (samples) is small, so it should just need a table scan.   I will test further using the sgdd file.